.github/workflows/rust.yml

```diff
@@ -1,15 +1,19 @@
-on: [push]
-
 name: Rust-CI
 
+on:
+  push:
+    paths:
+      - 'rust/**'
```

Comment:
Decided to change this to only run this extensive CI for changes inside the `rust` folder instead of doing it for *every* push. Makes much more sense.